packages/obelisk/obelisk.0.7.0/opam

```diff
@@ -0,0 +1,41 @@
+# This file is generated by dune, edit dune-project instead
+opam-version: "2.0"
+synopsis: "Pretty-printing for Menhir files"
+description: """
+Obelisk is a simple tool which produces pretty-printed output from a Menhir parser file (.mly).
+It is inspired from yacc2latex and is also written in OCaml, but is aimed at supporting features from Menhir instead of only those of ocamlyacc."""
+maintainer: ["Lélio Brun <lb@leliobrun.net>"]
+authors: ["Lélio Brun"]
+license: "MIT"
+homepage: "https://github.com/Lelio-Brun/Obelisk"
+doc: "https://github.com/Lelio-Brun/Obelisk/blob/master/README.md"
+bug-reports: "https://github.com/Lelio-Brun/obelisk/issues"
+depends: [
+  "ocaml" {>= "4.08"}
+  "dune" {>= "2.2.0"}
+  "re"
+  "menhir"
```

Comment:
```suggestion
  "menhir" {>= "20180523"}
```

Reply:
or rather matching the actual release

```suggestion
  "menhir" {>= "20180528"}
```

Reply:
Hello, this is strange, Dune is supposed to impose that dependency: 
https://github.com/ocaml/dune/blob/1e8b005dfe7d852887e0eb1d32e0260c9ef70d31/src/dune_rules/menhir/menhir_rules.ml#L10

But I can add the bound anyway, how should I do it, patch the opam file in my repo, and redo the whole `opam-publish` process with the updated release?

Reply:
From the [log](https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/fdb9880751944a2d3f44a2d33c4704f94c38e7bb/variant/compilers,4.14,obelisk.0.7.0,lower-bounds), I can't understand what triggers the downgrading after the successful dependencies installation: 
```
obelisk.0.7.0 is not installed. Install it? [y/n] y
The following actions will be performed:
=== install 8 packages
  - install dune      3.16.0         [required by obelisk]
  - install menhir    20231231       [required by obelisk]
  - install menhirCST 20231231       [required by menhir]
  - install menhirLib 20231231       [required by menhir]
  - install menhirSdk 20231231       [required by menhir]
  - install obelisk   0.7.0 (pinned)
  - install re        1.11.0         [required by obelisk]
  - install seq       base           [required by re]

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
-> retrieved dune.3.16.0  (cached)
-> retrieved menhir.20231231, menhirCST.20231231, menhirLib.20231231, menhirSdk.20231231  (cached)
-> retrieved obelisk.0.7.0  (cached)
-> retrieved re.1.11.0  (cached)
-> retrieved seq.base  (cached)
-> installed seq.base
-> installed dune.3.16.0
-> installed menhirCST.20231231
-> installed menhirSdk.20231231
-> installed menhirLib.20231231
-> installed re.1.11.0
-> installed menhir.20231231
-> installed obelisk.0.7.0
Done.
``` 

Why does the subsequent downgrading occur? It does not on my own CI (see the [log](https://github.com/Lelio-Brun/Obelisk/actions/runs/9766773894/job/26960481064)).

Reply:
- I encourage you to make the change upstream (i.e., to your repo) but you don't need to make a new release because it's only the packaging metadata which needs improving: you can just modify the files in this PR.
- The downgrading occurs as part of a special test in the opam-repository CI called "lowerbounds"; it's meant specifically to test the packages against the lowest bounds for their dependenvies constraints.

Reply:
Ok I see, thank you!